test/runtests.jl

```diff
@@ -745,6 +764,103 @@ const prob_pds_linmod_nonconservative_inplace = PDSProblem(linmodP!, linmodD!, [
                 end
             end
         end
+
+        # Check that the schemes accept zero initial values
+        @testset "Zero initial values" begin
+            # Do a single step and check that no NaNs occur
+            u0 = [1.0, 0.0]
+            dt = 1.0
+            tspan = (0.0, dt)
+            p = 1000.0
+            function prod!(P, u, p, t)
+                λ = p
+                fill!(P, zero(eltype(P)))
+                P[2, 1] = λ * u[1]
+            end
+            function dest!(D, u, p, t)
+                fill!(D, zero(eltype(D)))
+            end
+            function prod(u, p, t)
+                P = similar(u, (length(u), length(u)))
+                prod!(P, u, p, t)
+                return P
+            end
+            function dest(u, p, t)
+                d = similar(u)
+                dest!(d, u, p, t)
+                return d
+            end
+
+            prob_ip = ConservativePDSProblem(prod!, u0, tspan, p)
+            prob_ip_2 = PDSProblem(prod!, dest!, u0, tspan, p)
+            prob_oop = ConservativePDSProblem(prod, u0, tspan, p)
+            prob_oop_2 = PDSProblem(prod, dest, u0, tspan, p)
+
+            algs = (MPE(), MPRK22(0.5), MPRK22(1.0), MPRK22(2.0),
+                    MPRK43I(1.0, 0.5), MPRK43I(0.5, 0.75), MPRK43II(0.5),
+                    MPRK43II(2.0 / 3.0))
+
+            for alg in algs
+                sol = solve(prob_ip, alg; dt = dt, adaptive = false)
+                @test !any(isnan.(sol.u[end]))
+                sol = solve(prob_ip_2, alg; dt = dt, adaptive = false)
+                @test !any(isnan.(sol.u[end]))
+                sol = solve(prob_oop, alg; dt = dt, adaptive = false)
+                @test !any(isnan.(sol.u[end]))
+                sol = solve(prob_oop_2, alg; dt = dt, adaptive = false)
+                @test !any(isnan.(sol.u[end]))
```

Comment:
This version of `any` accepts a function as the first argument. This avoids compouting the full temporary vector of Bools and can exit early. This is not strictily necessary here since these tests are not critical for performance, but it's best practice to use it instead.

```suggestion
                sol = solve(prob_ip, alg; dt = dt, adaptive = false)
                @test !any(isnan, sol.u[end])
                sol = solve(prob_ip_2, alg; dt = dt, adaptive = false)
                @test !any(isnan, sol.u[end])
                sol = solve(prob_oop, alg; dt = dt, adaptive = false)
                @test !any(isnan, sol.u[end])
                sol = solve(prob_oop_2, alg; dt = dt, adaptive = false)
                @test !any(isnan, sol.u[end])
```